src/routes/posts.tsx

```diff
@@ -1,12 +1,31 @@
-import { PostItem } from "../components/Post/Post.tsx";
+import { PostItem } from "../components/Post/PostItem.tsx";
+import { config } from "../components/Post/post.config.ts";
+import { useGet } from "../hooks/useGet.ts";
+import { useParams } from "react-router-dom";
 import Post from "../components/Post/Post.types.ts";
 
-interface PostList {
-  postsList: Post[];
-}
+export function Posts() {
+  const { postsId } = useParams();
+  const { isLoading, isError, data } = useGet(config.url, postsId);
```

Comment:
Few things here:

`useGet` is a nice wrapper util, but to make it further useful, you can try something like we did in Xyte:
- There's a lib of its own named "api", which we import various react-query-based hooks from for various entities.
- Since this is not a monorepo with numerous libs, you can add a folder named `/api`, and there add:
  * `api.utils.ts` (instead of `useGet.ts`)
  * `posts.queries.ts` -> add all queries related to posts in there. So I'd expect to have there:
  
  ```
  export function usePosts() {
    return useGet(...)
  }
  ```
  
Which then you can import in `Posts` component:

```
import  { usePosts } from '../api/posts.queries.ts';

export function Posts() {
  ...
  
  const posts = usePosts();
  
  if (!posts.isFetched) ...
  if (posts.isError) ...
  
  return posts.data.map(...)
}
```

Then do same for other entities:

```
import { useUsers } from '../api/users.queries'
...
```

And better would be to add in `api/index.ts`:

```
export * from './posts.queries';
export * from ./users.queries';
```

Which then will allow you to import:

```
import  { usePosts, useUsers } from '.../api'
```